src/store/EncryptedStore.ts

```diff
@@ -122,10 +122,40 @@ export default class EncryptedKeyStore implements KeyStore {
     const secret = hexToBytes(
       await wallet.signMessage(storageSigRequestText(eBundle.walletPreKey))
     )
-    const ciphertext = new Ciphertext(eBundle.ciphertext)
-    const decrypted = await decrypt(ciphertext, secret)
-    const [bundle, needsUpdate2] = getPrivateBundle(decrypted)
-    return [bundle, needsUpdate || needsUpdate2]
+
+    // Ledger uses the last byte = v=[0,1,...] but Metamask and other wallets generate with
+    // v+27 as the last byte. We need to support both for interoperability. Doing this
+    // on the decryption side provides an immediate retroactive fix.
+    // Ledger is using the canonical way, whereas Ethereum adds 27 due to some legacy stuff
+    // https://github.com/ethereum/go-ethereum/issues/19751#issuecomment-504900739
+    try {
+      // Try the original version of the signature first
+      const ciphertext = new Ciphertext(eBundle.ciphertext)
+      const decrypted = await decrypt(ciphertext, secret)
+      const [bundle, needsUpdate2] = getPrivateBundle(decrypted)
+      return [bundle, needsUpdate || needsUpdate2]
+    } catch (e) {
+      // Assert that the secret is length 65 (encoded signature + recovery byte)
+      if (secret.length !== 65) {
+        throw new Error(
+          'Expected 65 bytes before trying a different recovery byte'
+        )
+      }
+      // Try the other version of recovery byte, either +27 or -27
+      const lastByte = secret[secret.length - 1]
+      let newSecret = secret.slice(0, secret.length - 1)
+      if (lastByte < 27) {
+        // This is a canonical signature, so we need to add 27 to the recovery byte and try again
+        newSecret = new Uint8Array([...newSecret, lastByte + 27])
+      } else {
+        // This canocalizes v to 0 or 1 (or maybe 2 or 3 but very unlikely)
+        newSecret = new Uint8Array([...newSecret, lastByte - 27])
+      }
```

Comment:
Is there a reason to do check for the signature format only on failure?
Can't we detect this case prior to decryption, and make the adjustment?

Reply:
I'm not sure if we can determine which case (0,1 or 27,28) was used for the original encryption operation without trying one first

Reply:
Ah I see. 

1) Can this be addressed in `toEncryptedBytes` as well standardizing our signature format would reduce complexity in the future. From a specification perspective its easier to say "EnableSignatures MUST be In X format" 
2) (OOS) but What would a migration look like for this? Is it possible to update the affected accounts in the future? so this code could be simplified?

Reply:
If we want to make changes to how we generate the secret, then I wouldn't pick v=0 or v=27, I'd just drop that byte altogether, especially since it's not entirely random.

Reply:
Yup, I'm tempted to make the `toEncryptedBytes` change separately, just to avoid too many moving pieces in a single PR. Your second point is a great one, we should be able to do a semi-transparent re-encryption the next time the user signs in